lib/trento/users.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Trento.Users do
+  @moduledoc """
+  The Users context.
+  """
+
+  use Pow.Ecto.Context,
+    repo: Trento.Repo,
+    user: Trento.Users.User
+
+  import Ecto.Query, warn: false
+  alias Trento.Repo
+
+  alias Trento.Users.User
+
+  @impl true
+  @doc """
+    get_by function overrides the one defined in Pow.Ecto.Context,
+    we retrieve the user by username as traditional Pow flow but we also exclude
+    deleted and locked users
+  """
+  def get_by(clauses) do
+    username = clauses[:username]
+
+    User
+    |> where([u], is_nil(u.deleted_at) and is_nil(u.locked_at) and u.username == ^username)
+    |> Repo.one()
+  end
+
+  def list_users do
+    User
+    |> where([u], is_nil(u.deleted_at))
+    |> Repo.all()
+  end
+
+  def get_user(id) do
```

Comment:
I guess this `get_user` is a function out of the standard `crud`, right?
To be used outside the controller functions.
Or maybe we don't need the `get_by` above.
Just asking hehe

Reply:
The function is an override of the Pow Context to customize the user retrieval in login